### PR TITLE
Update changelog and version.h for 5.12.0

### PR DESCRIPTION
**Pull request by absci**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,36 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](http://keepachangelog.com/)
 
+## 5.12.0 - 2024-01-31
+Updated PECL release packages. Here is the list of updates:
+
+### Added
+- Support for Alpine 3.19
```
Comment (David-Engel) [inline CHANGELOG.md:10]: Support for PHP 8.3?

Comment (absci) [reply CHANGELOG.md:10]: I added that when we did beta release [here](https://github.com/microsoft/msphpsql/pull/1502/files/e8e033bfdaeb885a76861d6c83cd6a7388242afc#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edL10). Should I include it also?

Comment (David-Engel) [reply CHANGELOG.md:10]: Yes. We should include all changes since 5.11.

Comment (absci) [reply CHANGELOG.md:11]: Sure, I updated the changelog.

**Review by David-Engel — APPROVED** [review]

**Pull request merged**